### PR TITLE
add event rule

### PR DESCRIPTION
**Pull request by yabloki**

### CODE REVIEW

File: src/UnderscoreEventArgs.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * @fileoverview Require that events arguments be prefixed by _
```
Comment (thorvald) [inline src/UnderscoreEventArgs.js]: not

File: src/UnderscoreEventArgs.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * @author Kirill Moizik <kirill@beam.io>
+ */
+const ruleId = 'underscore-events-args';
+const meta = {
+  type: 'naming',
+
+  docs: {
+    description: 'Event argument itentifiers must start with _',
```
Comment (thorvald) [inline src/UnderscoreEventArgs.js]: must not

File: src/UnderscoreEventArgs.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    category: 'Beam Style Guide Rules',
+  },
+
+  isDefault: true,
```
Comment (thorvald) [inline src/UnderscoreEventArgs.js]: This rule is default enabled, but recommended disabled?

File: test/rules/underscore-event-args.test.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    plugins: [
+      'beam',
+    ],
+    rules: {
+      'beam/underscore-events-args': 'error',
+    },
+  };
+
+  describe('with a event paramater that does not start with _', () => {
+    const code = contractWith('event badEvent(uint256 indexed _index, address _who);');
```
Comment (thorvald) [inline test/rules/underscore-event-args.test.js]: The description doesn't match the contract.

File: src/UnderscoreEventArgs.js (hunk 3 above)
Comment (yabloki) [reply src/UnderscoreEventArgs.js]: copy pasted from functions rule, fixed in both places

File: src/UnderscoreEventArgs.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * @fileoverview Require that events arguments not prefixed by _
+ * @author Kirill Moizik <kirill@beam.io>
+ */
+const ruleId = 'underscore-events-args';
```
Comment (mgregson) [inline src/UnderscoreEventArgs.js]: I think `no-underscore-event-args` is a bit more in line with the naming used for other rules.

**Pull request merged**